### DISCUSSION TITLE
Limiting Search to Specific Pages

### DISCUSSION THREAD

**Discussion by divdasani** [Pagefind/pagefind#721] [Q&A]
I'll preface by saying I have very little experience with frontend and JavaScript, so apologies for the noob questions!

I have a blog with pages that are specific to certain geographies (ie New York, London, etc). When a user visits a page within a certain geo, I would like the search on that page to also only display results within that geo.

1. Should I create separate indices and for each geo or use a shared index?
2. If it's better to use a shared index, how can I make the search filter to the given geo in its results without the user needing to do this manually?

Thank you :)

**Comment by bglw** [comment]
👋 Apologies for missing this!

I would recommend using a shared index. For the Default UI, you can set filters after initialization — see details for that here: https://pagefind.app/docs/ui-usage/#programmatically-controlling-the-pagefind-ui

The user would still be able to change the filtering, but the default would be correct per page :)

**Reply by divdasani** [reply]
Thanks for the response Liam! Is it possible to set filters programmatically with `triggerFilters` while disabling filtering within the UI?

**Reply by bglw** [reply]
You should be fine just adding the CSS somewhere:

```css
.pagefind-ui__filter-panel {
  display: none !important;
}
```
